cluster: vmselect query fails sporadically

**Describe the bug**

Sporadically query from the UI through vmselect fails and this is the log I see in vmselect
```
Aug 24 16:51:55 ip-172-30-89-153.ec2.internal vmselect[7433]: 2021-08-24T16:51:55.328Z        warn        VictoriaMetrics/app/vmselect/main.go:523        error in "/select/0/prometheus/api/v1/query?query=sort((time()%20-%20vm_app_start_timestamp%7Baws_name%3D~%22vm-agent-1%22%7D))&time=1629823916": error when executing query="sort((time() - vm_app_start_timestamp{aws_name=~\"vm-agent-1\"}))" for (time=1629823885328, step=300000): cannot evaluate "vm_app_start_timestamp{aws_name=~\"vm-agent-1\"}": error occured during search: cannot fetch query results from vmstorage nodes: cannot perform search on vmstorage 172.30.73.254:8401: cannot execute rpcName="search_v4" on vmstorage "172.30.73.254:8401" with timeout 30.000 seconds (elapsed 0.329 seconds); the timeout can be adjusted with `-search.maxQueryDuration` command-line flag: cannot read error message: cannot read 8 bytes with data size: cannot read data in 0.000 seconds: unexpected EOF; read only 0 bytes
```
Refreshing on grafana returns the query correctly.
At the same time, I have noticed that vmstorage remakes the connections to vmselect
```
Aug 24 16:51:55 ip-172-30-90-1.ec2.internal vmstorage[21193]: 2021-08-24T16:51:55.339Z        info        VictoriaMetrics/app/vmstorage/transport/server.go:164        accepted vmselect conn from 172.30.89.153:47206
Aug 24 16:51:55 ip-172-30-90-1.ec2.internal vmstorage[21193]: 2021-08-24T16:51:55.339Z        info        VictoriaMetrics/app/vmstorage/transport/server.go:164        accepted vmselect conn from 172.30.89.153:47212
Aug 24 16:51:55 ip-172-30-90-1.ec2.internal vmstorage[21193]: 2021-08-24T16:51:55.340Z        info        VictoriaMetrics/app/vmstorage/transport/server.go:164        accepted vmselect conn from 172.30.89.153:47216
Aug 24 16:51:55 ip-172-30-90-1.ec2.internal vmstorage[21193]: 2021-08-24T16:51:55.371Z        info        VictoriaMetrics/app/vmstorage/transport/server.go:164        accepted vmselect conn from 172.30.89.153:47218
Aug 24 16:51:55 ip-172-30-90-1.ec2.internal vmstorage[21193]: 2021-08-24T16:51:55.371Z        info        VictoriaMetrics/app/vmstorage/transport/server.go:207        processing vmselect conn from 172.30.89.153:47216
Aug 24 16:51:55 ip-172-30-90-1.ec2.internal vmstorage[21193]: 2021-08-24T16:51:55.371Z        info        VictoriaMetrics/app/vmstorage/transport/server.go:207        processing vmselect conn from 172.30.89.153:47206
Aug 24 16:51:55 ip-172-30-90-1.ec2.internal vmstorage[21193]: 2021-08-24T16:51:55.371Z        info        VictoriaMetrics/app/vmstorage/transport/server.go:207        processing vmselect conn from 172.30.89.153:47212
Aug 24 16:51:55 ip-172-30-90-1.ec2.internal vmstorage[21193]: 2021-08-24T16:51:55.383Z        info        VictoriaMetrics/app/vmstorage/transport/server.go:164        accepted vmselect conn from 172.30.89.153:47220
Aug 24 16:51:55 ip-172-30-90-1.ec2.internal vmstorage[21193]: 2021-08-24T16:51:55.398Z        info        VictoriaMetrics/app/vmstorage/transport/server.go:207        processing vmselect conn from 172.30.89.153:47218
Aug 24 16:51:55 ip-172-30-90-1.ec2.internal vmstorage[21193]: 2021-08-24T16:51:55.417Z        info        VictoriaMetrics/app/vmstorage/transport/server.go:207        processing vmselect conn from 172.30.89.153:47220
```


**Expected behavior**
No errors when querying the data

**Version**
```
vmselect-20210819-093240-tags-v1.64.1-cluster-0-gee1f3414d
```
```
vmstorage-20210819-093244-tags-v1.64.1-cluster-0-gee1f3414d
```


Let me know if there are any other stats that would help with this query. I am running the basic configs.

--- a/lib/netutil/conn_pool.go
+++ b/lib/netutil/conn_pool.go
@@ -71,7 +71,12 @@ func (cp *ConnPool) Get() (*handshake.BufferedConn, error) {
 //
 // Do not put broken and closed connections to the pool!
 func (cp *ConnPool) Put(bc *handshake.BufferedConn) {
-	bc.SetDeadline(time.Time{})
+	if err := bc.SetDeadline(time.Time{}); err != nil {
+		// Close the connection instead of returning it to the pool,
+		// since it may be broken.
+		_ = bc.Close()
+		return
+	}
 	cp.mu.Lock()
 	cp.conns = append(cp.conns, bc)
 	cp.mu.Unlock()

--- a/lib/netutil/conn_pool.go
+++ b/lib/netutil/conn_pool.go
@@ -3,6 +3,7 @@ package netutil
 import (
 	"fmt"
 	"sync"
+	"time"
 
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/handshake"
 )
@@ -70,6 +71,7 @@ func (cp *ConnPool) Get() (*handshake.BufferedConn, error) {
 //
 // Do not put broken and closed connections to the pool!
 func (cp *ConnPool) Put(bc *handshake.BufferedConn) {
+	bc.SetDeadline(time.Time{})
 	cp.mu.Lock()
 	cp.conns = append(cp.conns, bc)
 	cp.mu.Unlock()

